src/commentbot/github_utils.jl

```diff
@@ -0,0 +1,249 @@
+            rethrow(ex)
+        end
+    end
+
+    return nothing, nothing
+end
+
+is_owned_by_organization(event) = event.repository.owner.typ == "Organization"
+
+function is_comment_by_collaborator(event)
+    @debug("Checking if comment is by collaborator")
+    user = get_user_login(event.payload)
+    return iscollaborator(event.repository, user; auth=get_access_token(event))
+end
+
+function is_comment_by_org_owner_or_member(event)
+    @debug("Checking if comment is by repository parent organization owner or member")
+    org = event.repository.owner.login
+    user = get_user_login(event.payload)
+    if get(config["commentbot"], "check_private_membership", false)
+        return GitHub.check_membership(org, user; auth=get_user_auth())
+    else
+        return GitHub.check_membership(org, user; public_only=true)
+    end
+end
+
+has_register_rights(event) = is_comment_by_collaborator(event) || is_owned_by_organization(event) && is_comment_by_org_owner_or_member(event)
+
+is_pull_request(payload::Dict{<:AbstractString}) = haskey(payload, "pull_request") || haskey(payload, "issue") && haskey(payload["issue"], "pull_request")
+is_commit_comment(payload::Dict{<:AbstractString}) = haskey(payload, "comment") && !haskey(payload, "issue")
+
+function get_prid(payload::Dict{<:AbstractString})
+    if haskey(payload, "pull_request")
+        return payload["pull_request"]["number"]
+    elseif haskey(payload, "issue")
+        return payload["issue"]["number"]
+    else
+        error("Don't know how to get pull request number")
+    end
+end
+
+get_comment_commit_id(event) = event.payload["comment"]["commit_id"]
+get_clone_url(event) = event.payload["repository"]["clone_url"]
+
+function make_comment(evt::WebhookEvent, body::AbstractString)
+    config["commentbot"]["reply_comment"] || return
+    @debug("Posting comment to PR/issue")
+    headers = Dict("private_token" => config["github"]["token"])
+    params = Dict("body" => body)
+    repo = evt.repository
+    auth = get_user_auth()
+    if is_commit_comment(evt.payload)
+        GitHub.create_comment(repo, get_comment_commit_id(evt),
+                              :commit; headers=headers,
+                              params=params, auth=auth)
+    else
+        GitHub.create_comment(repo, get_prid(evt.payload),
+                              :issue; headers=headers,
+                              params=params, auth=auth)
+    end
+end
+
+function get_html_url(payload::Dict{<:AbstractString})
+    if haskey(payload, "pull_request")
+        return payload["pull_request"]["html_url"]
+    elseif haskey(payload, "issue")
+        if haskey(payload, "comment")
+            return payload["comment"]["html_url"]
+        else
+            return payload["issue"]["html_url"]
+        end
+    elseif haskey(payload, "comment")
+        return payload["comment"]["html_url"]
+    else
+        error("Don't know how to get html_url")
+    end
+end
+
+function raise_issue(event, phrase, bt)
```

Comment:
Docstring would bne nice here.
This does more things than I thought it would,
and the meaning of the  arg names are not clear to me